File: src/main/kotlin/com/stremio/core/types/addon/ResourcePath.kt
package com.stremio.core.types.addon

data class ResourcePath(
    val resource: String,
    val type: String,
    val id: String,
    val extra: List<ExtraValue>
) {
    companion object {
        const val META = "meta"
        const val STREAM = "stream"
        const val SUBTITLE = "subtitle"
    }
}

File: src/main/rust/bridge/resource_path.rs
use stremio_core::types::addon::ResourcePath;

use crate::bridge::{FromProtobuf, ToProtobuf};
use crate::protobuf::stremio::core::types;

impl FromProtobuf<ResourcePath> for types::ResourcePath {
    fn from_protobuf(&self) -> ResourcePath {
        ResourcePath {
            resource: self.resource.to_owned(),
            r#type: self.r#type.to_owned(),
            id: self.id.to_owned(),
            extra: self.extra.from_protobuf(),
        }
    }
}

impl ToProtobuf<types::ResourcePath, ()> for ResourcePath {
    fn to_protobuf(&self, _args: &()) -> types::ResourcePath {
        types::ResourcePath {
            resource: self.resource.to_owned(),
            r#type: self.r#type.to_owned(),
            id: self.id.to_owned(),
            extra: self.extra.to_protobuf(&()),
        }
    }
}
